test/jdk/java/lang/String/concat/HugeConcatTest.java

```diff
@@ -0,0 +1,74 @@
+/*
+ * Copyright (c) 2025, Oracle and/or its affiliates. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+/*
+ * @test
+ * @bug 8356152
+ * @summary Check that huge concatenations throw OutOfMemoryError
+ * @requires os.maxMemory > 6G
+ * @run junit/othervm -Xmx6G -XX:+CompactStrings -Dcompact=true HugeConcatTest
+ * @run junit/othervm -Xmx6G -XX:-CompactStrings -Dcompact=false HugeConcatTest
+ */
+
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Test;
+
+import static org.junit.Assert.assertThrows;
+
+public class HugeConcatTest {
```

Comment:
The same issue is reproduced with simple `+` concatenation.
```
        String r1 = "-".repeat(Integer.MAX_VALUE - 10);
        String r2 = "ы".repeat((Integer.MAX_VALUE - 10)  / 2);
        System.out.println(r1 + r2);
```
I think it makes sense to add related test for it too.

Reply:
@turbanoff I addressed your note.